src/dev/flang/ast/Expr.java

```diff
@@ -881,6 +696,149 @@ boolean isTypeAsValueCall()
   }
 
 
+  // NYI: CLEANUP: move this logic to isAssignableFrom?
+  /**
+   * check if assigning this expr to frmlT might
+   * be ambigous
+   *
+   * @param frmlT
+   */
+  void checkAmbiguousAssignmentToChoice(AbstractType frmlT)
+  {
+    var t = type();
+    if (frmlT.isChoice() && !t.isVoid() && frmlT.isAssignableFrom(t).yes())
+      {
+        AbstractType boxedType = boxedType(this, frmlT);
+        if (frmlT.isChoice() && frmlT.isAssignableFrom(boxedType).yes())
+          {
+            checkTagging(this, boxedType, frmlT);
+          }
+      }
+  }
+
+
+  /**
+   * @param expr the expr to be boxed
+   *
+   * @param frmlT the formal type
+   *
+   * @return the potentially boxed type
+   */
+  private static AbstractType boxedType(Expr expr, AbstractType frmlT)
```

Comment:
This should IMHO not be `static`, the `expr` argument should be the target `this`.